components/core/src/clp/EncodedVariableInterpreter.cpp

```diff
@@ -386,15 +386,31 @@ bool EncodedVariableInterpreter::encode_and_search_dictionary(
         LogTypeDictionaryEntry::add_float_var(logtype);
         sub_query.add_non_dict_var(encoded_var);
     } else {
-        auto entry = var_dict.get_entry_matching_value(var_str, ignore_case);
-        if (nullptr == entry) {
+        auto const entries = var_dict.get_entry_matching_value(var_str, ignore_case);
+
+        if (entries.empty()) {
             // Not in dictionary
             return false;
         }
-        encoded_var = encode_var_dict_id(entry->get_id());
 
         LogTypeDictionaryEntry::add_dict_var(logtype);
-        sub_query.add_dict_var(encoded_var, entry);
+
+        if (entries.size() == 1) {
+            auto const* entry = entries.at(0);
+            encoded_var = encode_var_dict_id(entry->get_id());
+            sub_query.add_dict_var(encoded_var, entry);
+            return true;
+        }
+
+        std::unordered_set<encoded_variable_t> encoded_vars;
+        std::unordered_set<clp::VariableDictionaryEntry const*> const entries_set(
+                entries.begin(),
+                entries.end()
+        );
+        for (auto const& entry : entries) {
+            encoded_vars.insert(encode_var_dict_id(entry->get_id()));
+        }
+        sub_query.add_imprecise_dict_var(encoded_vars, entries_set);
```

Comment:
Do you see any value turning this into a new method for SubQuery that adds imprecise_dict_var by taking in a ```vector<VariableDictionaryEntry*>```?

i.e.
```
void SubQuery::add_imprecise_dict_var(
        vector<VariableDictionaryEntry*> const& possible_var_dict_entries,
);
```